discover/portal/articles-dxp/190-managing-users/11-auditing-users/02-configuring-user-audits.markdown

```diff
@@ -0,0 +1,149 @@
+them for scheduled jobs, or disable them entirely. 
+
+There are two main ways to configure @product@:
+
+1.  Edit a configuration via Liferay's Control Panel and the configuration is
+    saved to Liferay's database.
+
+2.  Edit a configuration via an OSGi configuration file (`.config` file) in your
+    `[LIFERAY_HOME]/osgi/configs` folder.
+
+These methods apply to each of the audit configuration options explained below.
+
+## Reporting Audit Events in Liferay's Logs and Console [](id=reporting-audit-events-in-liferays-logs-and-console)
+
+**In the Control Panel:** Go to *Control Panel* &rarr; *Configuration* &rarr;
+*System Settings* and find the *Security* section. Click *Audit* &rarr; *Logging
+Message Audit Message Processor*.
+
+Check *Enabled* to report audit events in Liferay's log. Check *Output to
+Console* if you also want them in the console. Audit events are available in
+either CSV or JSON formats. Select the one you prefer and click *Save*.
+
+To report audit events in Liferay's logs and console via an OSGi configuration
+file, create a file called
+`com.liferay.portal.security.audit.router.configuration.LoggingAuditMessageProcessorConfiguration.config`
+in `[LIFERAY_HOME]/osgi/modules` folder containing these properties:
+
+	enabled="true"
+	logMessageFormat="CSV"
+	#logMessageFormat="JSON"
+	outputToConsole="true"
+
+As you can see, these are the same options from the Control Panel. Edit them
+you see fit.
+
+Regardless of configuration approach, you must also extend Liferay's
+`log4j-ext.xml` file to configure Log4J (@product@'s logging implementation) to
+log messages produced by the appropriate class to the appropriate file. To do
+so, create a `portal-log4j-ext.xml` file in 
+`[LIFERAY_HOME]/tomcat-[version]/webapps/ROOT/WEB-INF/classes/META-INF` 
+with this configuration:
+
+	<?xml version="1.0"?>
+	 <!DOCTYPE log4j:configuration SYSTEM "log4j.dtd">
+
+	<log4j:configuration xmlns:log4j="http://jakarta.apache.org/log4j/">
+
+		<!-- additional audit logging -->
+
+		<appender name="auditFile" class="org.apache.log4j.rolling.RollingFileAppender">
+			 <rollingPolicy class="org.apache.log4j.rolling.TimeBasedRollingPolicy">
+				 <param name="FileNamePattern" value="@liferay.home@/logs/audit.%d{yyyy-MM-dd}.log" />
+			 </rollingPolicy>
+			 <layout class="org.apache.log4j.EnhancedPatternLayout">
+				 <param name="ConversionPattern" value="%d{ABSOLUTE} %-5p [%t][%c{1}:%L] %m%n" />
+			 </layout>
+		 </appender>
+
+		<category name="com.liferay.portal.security.audit.router.internal.LoggingAuditMessageProcessor">
+			 <priority value="INFO" />
+			 <appender-ref ref="auditFile"/>
+		 </category>
+	</log4j:configuration>
+
+This configures Log4J to record INFO level messages from the
+`com.liferay.portal.security.audit.router.internal.LoggingAuditMessageProcessor`
+class to a file called `audit.yyyy-MM-dd.log` in the `[LIFERAY_HOME]/logs`
+folder. Adjust the audit file properties or log level to your liking.
+
```

Comment:
@shuyangzhou - Some questions about the log4j instructions above...

1. Is creating a `portal-log4j-ext.xml` file in the portal application's `WEB-INF/classes/META-INF/` folder appropriate? I thought you had to use an Ext plugin to extend portal's Log4j file (isn't portal's file called `portal-log4j.xml`?)

2. Is example `portal-log4j-ext.xml` code correct?

Thanks!